diff --git a/cmake/ValaPrecompile.cmake b/cmake/ValaPrecompile.cmake
index 6ab7fb2..3c75d40 100644
--- a/cmake/ValaPrecompile.cmake
+++ b/cmake/ValaPrecompile.cmake
@@ -27,7 +27,7 @@
 # either expressed or implied, of Jakob Westhoff
 ##
 
-include(ParseArguments)
+include(CMakeParseArguments)
 find_package(Vala REQUIRED)
 
 ##
@@ -114,7 +114,7 @@ find_package(Vala REQUIRED)
 ##
 
 macro(vala_precompile output target_name)
-    parse_arguments(ARGS "TARGET;PACKAGES;OPTIONS;DIRECTORY;GENERATE_GIR;GENERATE_SYMBOLS;GENERATE_HEADER;GENERATE_VAPI;CUSTOM_VAPIS" "" ${ARGN})
+    cmake_parse_arguments(ARGS "" "DIRECTORY;GENERATE_GIR;GENERATE_SYMBOLS;GENERATE_HEADER;GENERATE_VAPI" "PACKAGES;OPTIONS;CUSTOM_VAPIS" ${ARGN})
 
     if(ARGS_DIRECTORY)
         set(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/${ARGS_DIRECTORY})
@@ -130,7 +130,7 @@ macro(vala_precompile output target_name)
     set(out_files "")
     set(out_files_display "")
     set(${output} "")
-    foreach(src ${ARGS_DEFAULT_ARGS})
+    foreach(src ${ARGS_UNPARSED_ARGUMENTS})
         list(APPEND in_files "${CMAKE_CURRENT_SOURCE_DIR}/${src}")
         string(REPLACE ".vala" ".c" src ${src})
         string(REPLACE ".gs" ".c" src ${src})
diff --git a/cmake/ValaVersion.cmake b/cmake/ValaVersion.cmake
index 3fff193..c493a27 100644
--- a/cmake/ValaVersion.cmake
+++ b/cmake/ValaVersion.cmake
@@ -27,7 +27,7 @@
 # either expressed or implied, of Jakob Westhoff
 ##
 
-include(ParseArguments)
+include(CMakeParseArguments)
 find_package(Vala REQUIRED)
 
 ##
@@ -52,7 +52,7 @@ find_package(Vala REQUIRED)
 # If no option is specified the version will be treated as a minimal version.
 ##
 macro(ensure_vala_version version)
-    parse_arguments(ARGS "" "MINIMUM;MAXIMUM;EXACT" ${ARGN})
+    cmake_parse_arguments(ARGS "MINIMUM;MAXIMUM;EXACT" "" "" ${ARGN})
 	set(compare_message "")
 	set(error_message "")
 	if(ARGS_MINIMUM)
